packages/core/types/tests.tsx

```diff
@@ -123,3 +124,62 @@ interface TestTheme1 {
     )
   }}
 </ClassNames>
+
+// issue 1532 https://github.com/emotion-js/emotion/pull/1532
+interface HaveObjectCss<T = any> {
+  less: ObjectInterpolation<T>
+}
+// prior to 1532 the css prop being passed less would fail
+const LessComponent = <T extends any>({ less }: HaveObjectCss<T>) => (
+  <div css={less}>I have Css</div>
+)
+
+interface AnimationTheme {
+  from: { top: number }
+  to: { top: number }
+}
+
+// BEGIN NON TEMPLATE ANIMATIONS
+// without restricting less
+;<LessComponent
+  less={keyframes<AnimationTheme>({ from: { top: 0 }, to: { top: 20 } })}
+/>
+
+// FAILS IN TYPESCRIPT 2.8 only due to <LessComponent<AnimationTheme> not being supported
+// ;<LessComponent<AnimationTheme>
```

Comment:
If you want to do a breaking minor / major change to drop support for typescript 2.8, then these tests would pass.